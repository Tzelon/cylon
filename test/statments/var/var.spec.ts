import { loadTests } from '../../helpers';

describe('variable declerations', () => {
    loadTests(__dirname).forEach(test => {

        it(`should ${test.filename}`, () => {
            expect(test.compiled.code).toMatchSnapshot();
            expect(test.compiled.parse).toMatchSnapshot();
        })
    });
});
