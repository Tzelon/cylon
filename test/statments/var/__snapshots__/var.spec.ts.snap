// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`variable declerations should declare-function.cy 1`] = `
"import $NEO from \\"./neo.runtime.js\\"

var func = $NEO.stone(function () {
    return \\"hello world\\";
});"
`;

exports[`variable declerations should declare-function.cy 2`] = `
Object {
  "id": "",
  "scope": Object {
    "func": Object {
      "alphameric": true,
      "column_nr": 4,
      "column_to": 8,
      "id": "func",
      "line_nr": 0,
      "origin": [Circular],
      "readonly": true,
    },
  },
  "zeroth": Array [
    Object {
      "alphameric": true,
      "class": "statement",
      "column_nr": 0,
      "column_to": 3,
      "id": "def",
      "line_nr": 0,
      "wunth": Object {
        "column_nr": 10,
        "column_to": 11,
        "id": "Æ’",
        "line_nr": 0,
        "parent": [Circular],
        "scope": Object {},
        "wunth": Array [
          Object {
            "alphameric": true,
            "class": "statement",
            "column_nr": 4,
            "column_to": 10,
            "disrupt": true,
            "id": "return",
            "line_nr": 1,
            "return": true,
            "zeroth": Object {
              "column_nr": 11,
              "column_to": 24,
              "id": "(text)",
              "line_nr": 1,
              "readonly": true,
              "text": "hello world",
            },
          },
        ],
        "zeroth": Array [],
      },
      "zeroth": Object {
        "alphameric": true,
        "column_nr": 4,
        "column_to": 8,
        "id": "func",
        "line_nr": 0,
        "origin": [Circular],
        "readonly": true,
      },
    },
  ],
}
`;

exports[`variable declerations should declare-number.cy 1`] = `
"import $NEO from \\"./neo.runtime.js\\"
const $10000 = $NEO.number(\\"10000\\");

var variable = $10000;"
`;

exports[`variable declerations should declare-number.cy 2`] = `
Object {
  "id": "",
  "scope": Object {
    "variable": Object {
      "alphameric": true,
      "column_nr": 4,
      "column_to": 12,
      "id": "variable",
      "line_nr": 0,
      "origin": [Circular],
      "readonly": false,
    },
  },
  "zeroth": Array [
    Object {
      "alphameric": true,
      "class": "statement",
      "column_nr": 0,
      "column_to": 3,
      "id": "var",
      "line_nr": 0,
      "wunth": Object {
        "column_nr": 14,
        "column_to": 19,
        "id": "(number)",
        "line_nr": 0,
        "number": Object {
          "coefficient": Array [
            "+",
            10000,
          ],
          "exponent": 0,
        },
        "readonly": true,
        "text": "10000",
      },
      "zeroth": Object {
        "alphameric": true,
        "column_nr": 4,
        "column_to": 12,
        "id": "variable",
        "line_nr": 0,
        "origin": [Circular],
        "readonly": false,
      },
    },
  ],
}
`;

exports[`variable declerations should declare-text.cy 1`] = `
"import $NEO from \\"./neo.runtime.js\\"

var text = \\"Hello World\\";"
`;

exports[`variable declerations should declare-text.cy 2`] = `
Object {
  "id": "",
  "scope": Object {
    "text": Object {
      "alphameric": true,
      "column_nr": 4,
      "column_to": 8,
      "id": "text",
      "line_nr": 0,
      "origin": [Circular],
      "readonly": false,
    },
  },
  "zeroth": Array [
    Object {
      "alphameric": true,
      "class": "statement",
      "column_nr": 0,
      "column_to": 3,
      "id": "var",
      "line_nr": 0,
      "wunth": Object {
        "column_nr": 10,
        "column_to": 23,
        "id": "(text)",
        "line_nr": 0,
        "readonly": true,
        "text": "Hello World",
      },
      "zeroth": Object {
        "alphameric": true,
        "column_nr": 4,
        "column_to": 8,
        "id": "text",
        "line_nr": 0,
        "origin": [Circular],
        "readonly": false,
      },
    },
  ],
}
`;
